### PR TITLE
Finish APIs for Authentication & Authorization

### PR DESCRIPTION
**Pull request by nguyenquocviet2007**
- create APIs for user:
  1. register
  2. login

### CODE REVIEW

File: server/controller/Users/UserQueries.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const dbconnection = require('../../database/db_connection');
+
+const mysql = require("mysql");
+    // TODO: Global database config
+    const dbConfig = mysql.createConnection({
+        user: "b4bb330a3f009b",
+        host: "eu-cdbr-west-01.cleardb.com",
+        password: "4be56a74",
+        database: "heroku_1fcc54407741c0e",
+      });
```
Comment (khanhphd-student) [inline server/controller/Users/UserQueries.js:10]: This config has been set up in the global file.

Comment (TrNgTien) [reply server/controller/Users/UserQueries.js:10]: Yesterday, I require him to demo for me, then the config has to put in there to test, I told him has to refactor to match with the global config @Dev01-vinamr-Ken

**Comment by TrNgTien** [comment]
Have to refactor codes to match with global db config @nguyenquocviet2007

File: server/controller/Users/UserQueries.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+module.exports = {
+    register: (data, callBack) => {
+        dbConfig.query(
+            `insert into users(username, password) 
+                        values(?,?)`,
+            [
+                data.username,
+                data.password
```
Comment (TrNgTien) [inline server/controller/Users/UserQueries.js:19]: I think @nguyenquocviet2007  you should have I more query is insert into userstype and insert the id with randomInt, I think @Dev01-vinamr-Ken did it!

File: server/controller/Users/UserQueries.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        user: "b4bb330a3f009b",
+        host: "eu-cdbr-west-01.cleardb.com",
+        password: "4be56a74",
+        database: "heroku_1fcc54407741c0e",
+      });
+
+module.exports = {
+    register: (data, callBack) => {
+        dbConfig.query(
+            `insert into users(username, password)
```
Comment (TrNgTien) [inline server/controller/Users/UserQueries.js:15]: please write the queries with the capitalize such as INSERT INTO ..... @nguyenquocviet2007

**Pull request merged**